mpsse_darwin.go

```diff
@@ -550,8 +517,14 @@ func (m *Mpsse) Read(size int) string {
 	// given number of times - if a result comes back as 0x00, we do
 	// not treat it as empty, but instead add it to the response string
 	// as the value \x00.
+
+	// Above means that golang is treating "\x00" as an empty C
+	// null terminated string which makes sense. Here this is just
+	// bytes, and we want to keep the \x00 bytes.
 	for i := 0; i < size; i++ {
-		read := C.GoString(C.Read(m.ctx, C.int(1)))
+		charPtr := C.Read(m.ctx, C.int(1))
```

Comment:
This is the simple read leak fix. In general, but not necessarily always, when a C function returns a pointer it's up to the caller to free it.

Reply:
ah, this is good to know. good catch.

its been a looooong while since I did anything with C (and even then I didn't do much with it), so I'm not surprised I didn't now this